pkg/controller/discovery/model/event.go

```diff
@@ -0,0 +1,93 @@
+package model
+
+import (
+	"encoding/json"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+//
+// Event model.
+type Event struct {
+	Base
+}
+
+//
+// Update the model `with` a k8s Event.
+func (m *Event) With(object *v1.Event) {
+	m.UID = string(object.UID)
+	m.Version = object.ResourceVersion
+	m.Namespace = object.Namespace
+	m.Name = object.Name
+	m.labels = Labels{
+		"involvedObjectKind":      object.InvolvedObject.Kind,
```

Comment:
Events don't have labels, so I'm deriving some based on the `involvedObject` field on the event to allow us to query for events related to a particular resource.